elixir/apps/web/lib/web/controllers/auth_controller.ex

```diff
@@ -353,7 +355,12 @@ defmodule Web.AuthController do
     )
   end
 
-  defp put_if_not_nil(map \\ %{}, key, value)
-  defp put_if_not_nil(map, _key, nil), do: map
-  defp put_if_not_nil(map, key, value), do: Map.put(map, key, value)
+  defp take_non_empty_params(map, keys) do
+    map |> Map.take(keys) |> Map.reject(fn {_key, value} -> value in ["", nil] end)
```

Comment:
Wondering if there should also be a `trim` call in here?  Do we want to allow a string only made up of spaces?

Reply:
Ok, sounds good.  Just wanted to make sure there weren't any negative consequences for a string made up of spaces.

Reply:
It will break the auth flow for a mobile client but we don't even show the address string there, so no point to optimize for it

Reply:
Gotcha....as long as it fails in a secure way (i.e. doesn't authenticate) then it sounds good to me.